autonomy/cli/scaffold_fsm_templates/components.py

```diff
@@ -0,0 +1,379 @@
+    from packages.valory.skills.abstract_round_abci.behaviours import (
+        AbstractRoundBehaviour,
+        BaseBehaviour,
+    )
+
+    from packages.{author}.skills.{skill_name}.models import Params
+    from packages.{author}.skills.{skill_name}.rounds import (
+        SynchronizedData,
+        {AbciApp},
+        {rounds},
+    )
+
+    """
+
+    BASE_BEHAVIOUR_CLS = """\
+    class {BaseBehaviourCls}(BaseBehaviour):
+        \"\"\"Base behaviour for the common apps' skill.\"\"\"
+
+        @property
+        def synchronized_data(self) -> SynchronizedData:
+            \"\"\"Return the synchronized data.\"\"\"
+            return cast(SynchronizedData, super().synchronized_data)
+
+        @property
+        def params(self) -> Params:
+            \"\"\"Return the params.\"\"\"
+            return cast(Params, super().params)
+
+    """
+
+    BEHAVIOUR_CLS = """\
+    class {BehaviourCls}({BaseBehaviourCls}):
+        \"\"\"{BehaviourCls}\"\"\"
+
+        # TODO: set the following class attributes
+        state_id: str
+        behaviour_id: str = "{behaviour_id}"
+        matching_round: Type[AbstractRound] = {matching_round}
+
+        @abstractmethod
+        def async_act(self) -> Generator:
+            \"\"\"Do the act, supporting asynchronous execution.\"\"\"
+
+    """
+
+    ROUND_BEHAVIOUR_CLS = """\
+    class {RoundBehaviourCls}(AbstractRoundBehaviour):
+        \"\"\"{RoundBehaviourCls}\"\"\"
+
+        initial_behaviour_cls = {InitialBehaviourCls}
+        abci_app_cls = {AbciApp}  # type: ignore
+        behaviours: Set[Type[BaseBehaviour]] = {round_behaviours}
+    """
+
+
+@dataclass
+class PAYLOADS:
+    """Template for payloads.py"""
+
+    FILENAME = "payloads.py"
+
+    HEADER = """\
+    \"\"\"This module contains the transaction payloads of the {AbciApp}.\"\"\"
+
+    from abc import ABC
+    from enum import Enum
+    from typing import Any, Dict, Hashable, Optional
+
+    from packages.valory.skills.abstract_round_abci.base import BaseTxPayload
+
+    """
+
+    TRANSACTION_TYPE_SECTION = """\
+    class TransactionType(Enum):
+        \"\"\"Enumeration of transaction types.\"\"\"
+
+        # TODO: define transaction types: e.g. TX_HASH: "tx_hash"
+        {tx_types}
+
+        def __str__(self) -> str:
+            \"\"\"Get the string value of the transaction type.\"\"\"
+            return self.value
+
+    """
+
+    BASE_PAYLOAD_CLS = """\
+    class Base{FSMName}Payload(BaseTxPayload, ABC):
+        \"\"\"Base payload for {AbciApp}.\"\"\"
+
+        def __init__(self, sender: str, content: Hashable, **kwargs: Any) -> None:
+            \"\"\"Initialize a 'select_keeper' transaction payload.\"\"\"
+
+            super().__init__(sender, **kwargs)
+            setattr(self, f"_{{self.transaction_type}}", content)
+            p = property(lambda s: getattr(self, f"_{{self.transaction_type}}"))
+            setattr(self.__class__, f"{{self.transaction_type}}", p)
+
```

Comment:
That's nice!

Reply:
thanks for comment. Because now I noticed the docstring is incorrect :smiling_face_with_tear: